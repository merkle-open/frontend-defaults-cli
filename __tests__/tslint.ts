import { defaultIt } from './shared';

describe('tslint', async () => {
	it('default', async () => await defaultIt('tslint', '--tslint --force --noInstall'));
	it('with-prettier', async () => await defaultIt('tslint-with-prettier', '--tslint --prettier --force --noInstall'));
	it('with-githooks', async () => await defaultIt('tslint-with-githooks', '--tslint --githooks --force --noInstall'));
});
