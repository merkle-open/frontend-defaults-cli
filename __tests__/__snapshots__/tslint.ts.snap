// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tslint default 1`] = `"{  \\"options\\": {    \\"ts\\": false,    \\"tslint\\": true,    \\"es\\": false,    \\"eslint\\": false,    \\"editorconfig\\": false,    \\"prettier\\": false,    \\"stylelint\\": false,    \\"gitignore\\": false,    \\"npmrc\\": false,    \\"readme\\": false,    \\"githooks\\": false,    \\"commitlint\\": false,    \\"nodeVersion\\": false,    \\"webpack\\": false,    \\"install\\": false,    \\"force\\": true,    \\"dryRun\\": false,    \\"mode\\": \\"api\\"  },  \\"mergedFiles\\": {    \\"package.json\\": {      \\"override\\": false,      \\"data\\": {        \\"scripts\\": {          \\"lint\\": \\"npm-run-all lint:*\\",          \\"lint:ts\\": \\"tslint 'src/**/*.{ts,tsx}'\\"        },        \\"devDependencies\\": {          \\"@namics/tslint-config\\": \\"0.2.0\\",          \\"npm-run-all\\": \\"4.1.5\\",          \\"tslint\\": \\"5.12.1\\"        }      }    },    \\"tslint.json\\": {      \\"override\\": false,      \\"data\\": \\"{\\\\n\\\\t\\\\\\"defaultSeverity\\\\\\": \\\\\\"error\\\\\\",\\\\n\\\\t\\\\\\"extends\\\\\\": [\\\\\\"@namics/tslint-config\\\\\\"],\\\\n\\\\t\\\\\\"jsRules\\\\\\": {},\\\\n\\\\t\\\\\\"rules\\\\\\": {},\\\\n\\\\t\\\\\\"rulesDirectory\\\\\\": []\\\\n}\\\\n\\"    }  }}"`;

exports[`tslint default 2`] = `"{  \\"scripts\\": {    \\"lint\\": \\"npm-run-all lint:*\\",    \\"lint:ts\\": \\"tslint 'src/**/*.{ts,tsx}'\\"  },  \\"devDependencies\\": {    \\"@namics/tslint-config\\": \\"0.2.0\\",    \\"npm-run-all\\": \\"4.1.5\\",    \\"tslint\\": \\"5.12.1\\"  }}"`;

exports[`tslint default 3`] = `"{	\\"defaultSeverity\\": \\"error\\",	\\"extends\\": [\\"@namics/tslint-config\\"],	\\"jsRules\\": {},	\\"rules\\": {},	\\"rulesDirectory\\": []}"`;

exports[`tslint with-githooks 1`] = `"{  \\"options\\": {    \\"ts\\": false,    \\"tslint\\": true,    \\"es\\": false,    \\"eslint\\": false,    \\"editorconfig\\": false,    \\"prettier\\": false,    \\"stylelint\\": false,    \\"gitignore\\": false,    \\"npmrc\\": false,    \\"readme\\": false,    \\"githooks\\": true,    \\"commitlint\\": false,    \\"nodeVersion\\": false,    \\"webpack\\": false,    \\"install\\": false,    \\"force\\": true,    \\"dryRun\\": false,    \\"mode\\": \\"api\\"  },  \\"mergedFiles\\": {    \\"package.json\\": {      \\"override\\": false,      \\"data\\": {        \\"scripts\\": {          \\"lint\\": \\"npm-run-all lint:*\\",          \\"lint:ts\\": \\"tslint 'src/**/*.{ts,tsx}'\\"        },        \\"devDependencies\\": {          \\"@namics/tslint-config\\": \\"0.2.0\\",          \\"husky\\": \\"1.2.0\\",          \\"lint-staged\\": \\"8.1.0\\",          \\"npm-run-all\\": \\"4.1.5\\",          \\"tslint\\": \\"5.12.1\\"        },        \\"husky\\": {          \\"hooks\\": {            \\"pre-commit\\": \\"lint-staged\\"          }        },        \\"lint-staged\\": {          \\"linters\\": {            \\"src/**/*.{ts,tsx}\\": [              \\"tslint\\"            ]          }        }      }    },    \\".huskyrc\\": {      \\"override\\": false,      \\"data\\": \\"# ~/.huskyrc\\\\n# Note: it will run in /bin/sh shell\\\\nif command -v nodenv >/dev/null 2>&1; then\\\\n  eval \\\\\\"$(nodenv init -)\\\\\\"\\\\nfi\\"    },    \\"tslint.json\\": {      \\"override\\": false,      \\"data\\": \\"{\\\\n\\\\t\\\\\\"defaultSeverity\\\\\\": \\\\\\"error\\\\\\",\\\\n\\\\t\\\\\\"extends\\\\\\": [\\\\\\"@namics/tslint-config\\\\\\"],\\\\n\\\\t\\\\\\"jsRules\\\\\\": {},\\\\n\\\\t\\\\\\"rules\\\\\\": {},\\\\n\\\\t\\\\\\"rulesDirectory\\\\\\": []\\\\n}\\\\n\\"    }  }}"`;

exports[`tslint with-githooks 2`] = `"# ~/.huskyrc# Note: it will run in /bin/sh shellif command -v nodenv >/dev/null 2>&1; then  eval \\"$(nodenv init -)\\"fi"`;

exports[`tslint with-githooks 3`] = `"{  \\"scripts\\": {    \\"lint\\": \\"npm-run-all lint:*\\",    \\"lint:ts\\": \\"tslint 'src/**/*.{ts,tsx}'\\"  },  \\"devDependencies\\": {    \\"@namics/tslint-config\\": \\"0.2.0\\",    \\"husky\\": \\"1.2.0\\",    \\"lint-staged\\": \\"8.1.0\\",    \\"npm-run-all\\": \\"4.1.5\\",    \\"tslint\\": \\"5.12.1\\"  },  \\"husky\\": {    \\"hooks\\": {      \\"pre-commit\\": \\"lint-staged\\"    }  },  \\"lint-staged\\": {    \\"linters\\": {      \\"src/**/*.{ts,tsx}\\": [        \\"tslint\\"      ]    }  }}"`;

exports[`tslint with-githooks 4`] = `"{	\\"defaultSeverity\\": \\"error\\",	\\"extends\\": [\\"@namics/tslint-config\\"],	\\"jsRules\\": {},	\\"rules\\": {},	\\"rulesDirectory\\": []}"`;

exports[`tslint with-prettier 1`] = `"{  \\"options\\": {    \\"ts\\": false,    \\"tslint\\": true,    \\"es\\": false,    \\"eslint\\": false,    \\"editorconfig\\": false,    \\"prettier\\": true,    \\"stylelint\\": false,    \\"gitignore\\": false,    \\"npmrc\\": false,    \\"readme\\": false,    \\"githooks\\": false,    \\"commitlint\\": false,    \\"nodeVersion\\": false,    \\"webpack\\": false,    \\"install\\": false,    \\"force\\": true,    \\"dryRun\\": false,    \\"mode\\": \\"api\\"  },  \\"mergedFiles\\": {    \\"package.json\\": {      \\"override\\": false,      \\"data\\": {        \\"scripts\\": {          \\"lint\\": \\"npm-run-all lint:*\\",          \\"lint:ts\\": \\"tslint 'src/**/*.{ts,tsx}'\\",          \\"prettier\\": \\"prettier --write \\\\\\"**/*.*(js|jsx|ts|tsx|json)\\\\\\"\\"        },        \\"devDependencies\\": {          \\"@namics/prettier-config\\": \\"0.2.0\\",          \\"@namics/tslint-config\\": \\"0.2.0\\",          \\"npm-run-all\\": \\"4.1.5\\",          \\"prettier\\": \\"1.15.2\\",          \\"tslint\\": \\"5.12.1\\",          \\"tslint-config-prettier\\": \\"1.17.0\\"        }      }    },    \\".prettierignore\\": {      \\"override\\": false,      \\"data\\": \\"# Node\\\\n**/node_modules\\\\n**/package.json\\\\n**/package-lock.json\\\\n\\\\n# Generated\\\\n**/bower.json\\\\n**/lerna.json\\\\n\\\\n# Build\\\\n**/build\\\\n**/dist\\\\n**/public\\\\n**/coverage\\\\n**/storybook-static\\\\n\\\\n# Nitro\\\\n**/project/blueprints\\\\n**/.yo-rc.json\\"    },    \\".prettierrc.js\\": {      \\"override\\": false,      \\"data\\": \\"module.exports = require('@namics/prettier-config');\\\\n\\"    },    \\"tslint.json\\": {      \\"override\\": false,      \\"data\\": \\"{\\\\n\\\\t\\\\\\"defaultSeverity\\\\\\": \\\\\\"error\\\\\\",\\\\n\\\\t\\\\\\"extends\\\\\\": [\\\\\\"@namics/tslint-config\\\\\\", \\\\\\"tslint-config-prettier\\\\\\"],\\\\n\\\\t\\\\\\"jsRules\\\\\\": {},\\\\n\\\\t\\\\\\"rules\\\\\\": {},\\\\n\\\\t\\\\\\"rulesDirectory\\\\\\": []\\\\n}\\\\n\\"    }  }}"`;

exports[`tslint with-prettier 2`] = `"# Node**/node_modules**/package.json**/package-lock.json# Generated**/bower.json**/lerna.json# Build**/build**/dist**/public**/coverage**/storybook-static# Nitro**/project/blueprints**/.yo-rc.json"`;

exports[`tslint with-prettier 3`] = `"module.exports = require('@namics/prettier-config');"`;

exports[`tslint with-prettier 4`] = `"{  \\"scripts\\": {    \\"lint\\": \\"npm-run-all lint:*\\",    \\"lint:ts\\": \\"tslint 'src/**/*.{ts,tsx}'\\",    \\"prettier\\": \\"prettier --write \\\\\\"**/*.*(js|jsx|ts|tsx|json)\\\\\\"\\"  },  \\"devDependencies\\": {    \\"@namics/prettier-config\\": \\"0.2.0\\",    \\"@namics/tslint-config\\": \\"0.2.0\\",    \\"npm-run-all\\": \\"4.1.5\\",    \\"prettier\\": \\"1.15.2\\",    \\"tslint\\": \\"5.12.1\\",    \\"tslint-config-prettier\\": \\"1.17.0\\"  }}"`;

exports[`tslint with-prettier 5`] = `"{	\\"defaultSeverity\\": \\"error\\",	\\"extends\\": [\\"@namics/tslint-config\\", \\"tslint-config-prettier\\"],	\\"jsRules\\": {},	\\"rules\\": {},	\\"rulesDirectory\\": []}"`;

exports[`tslint with-prettier-githooks 1`] = `"{  \\"options\\": {    \\"ts\\": false,    \\"tslint\\": true,    \\"es\\": false,    \\"eslint\\": false,    \\"editorconfig\\": false,    \\"prettier\\": true,    \\"stylelint\\": false,    \\"gitignore\\": false,    \\"npmrc\\": false,    \\"readme\\": false,    \\"githooks\\": true,    \\"commitlint\\": false,    \\"nodeVersion\\": false,    \\"webpack\\": false,    \\"install\\": false,    \\"force\\": true,    \\"dryRun\\": false,    \\"mode\\": \\"api\\"  },  \\"mergedFiles\\": {    \\"package.json\\": {      \\"override\\": false,      \\"data\\": {        \\"lint-staged\\": {          \\"linters\\": {            \\"*.{js,jsx,ts,tsx,json}\\": [              \\"prettier --list-different --write\\"            ],            \\"src/**/*.{ts,tsx}\\": [              \\"tslint\\"            ]          }        },        \\"scripts\\": {          \\"lint\\": \\"npm-run-all lint:*\\",          \\"lint:ts\\": \\"tslint 'src/**/*.{ts,tsx}'\\",          \\"prettier\\": \\"prettier --write \\\\\\"**/*.*(js|jsx|ts|tsx|json)\\\\\\"\\"        },        \\"husky\\": {          \\"hooks\\": {            \\"pre-commit\\": \\"lint-staged\\"          }        },        \\"devDependencies\\": {          \\"@namics/prettier-config\\": \\"0.2.0\\",          \\"@namics/tslint-config\\": \\"0.2.0\\",          \\"husky\\": \\"1.2.0\\",          \\"lint-staged\\": \\"8.1.0\\",          \\"npm-run-all\\": \\"4.1.5\\",          \\"prettier\\": \\"1.15.2\\",          \\"tslint\\": \\"5.12.1\\",          \\"tslint-config-prettier\\": \\"1.17.0\\"        }      }    },    \\".huskyrc\\": {      \\"override\\": false,      \\"data\\": \\"# ~/.huskyrc\\\\n# Note: it will run in /bin/sh shell\\\\nif command -v nodenv >/dev/null 2>&1; then\\\\n  eval \\\\\\"$(nodenv init -)\\\\\\"\\\\nfi\\"    },    \\".prettierignore\\": {      \\"override\\": false,      \\"data\\": \\"# Node\\\\n**/node_modules\\\\n**/package.json\\\\n**/package-lock.json\\\\n\\\\n# Generated\\\\n**/bower.json\\\\n**/lerna.json\\\\n\\\\n# Build\\\\n**/build\\\\n**/dist\\\\n**/public\\\\n**/coverage\\\\n**/storybook-static\\\\n\\\\n# Nitro\\\\n**/project/blueprints\\\\n**/.yo-rc.json\\"    },    \\".prettierrc.js\\": {      \\"override\\": false,      \\"data\\": \\"module.exports = require('@namics/prettier-config');\\\\n\\"    },    \\"tslint.json\\": {      \\"override\\": false,      \\"data\\": \\"{\\\\n\\\\t\\\\\\"defaultSeverity\\\\\\": \\\\\\"error\\\\\\",\\\\n\\\\t\\\\\\"extends\\\\\\": [\\\\\\"@namics/tslint-config\\\\\\", \\\\\\"tslint-config-prettier\\\\\\"],\\\\n\\\\t\\\\\\"jsRules\\\\\\": {},\\\\n\\\\t\\\\\\"rules\\\\\\": {},\\\\n\\\\t\\\\\\"rulesDirectory\\\\\\": []\\\\n}\\\\n\\"    }  }}"`;

exports[`tslint with-prettier-githooks 2`] = `"# ~/.huskyrc# Note: it will run in /bin/sh shellif command -v nodenv >/dev/null 2>&1; then  eval \\"$(nodenv init -)\\"fi"`;

exports[`tslint with-prettier-githooks 3`] = `"# Node**/node_modules**/package.json**/package-lock.json# Generated**/bower.json**/lerna.json# Build**/build**/dist**/public**/coverage**/storybook-static# Nitro**/project/blueprints**/.yo-rc.json"`;

exports[`tslint with-prettier-githooks 4`] = `"module.exports = require('@namics/prettier-config');"`;

exports[`tslint with-prettier-githooks 5`] = `"{  \\"lint-staged\\": {    \\"linters\\": {      \\"*.{js,jsx,ts,tsx,json}\\": [        \\"prettier --list-different --write\\"      ],      \\"src/**/*.{ts,tsx}\\": [        \\"tslint\\"      ]    }  },  \\"scripts\\": {    \\"lint\\": \\"npm-run-all lint:*\\",    \\"lint:ts\\": \\"tslint 'src/**/*.{ts,tsx}'\\",    \\"prettier\\": \\"prettier --write \\\\\\"**/*.*(js|jsx|ts|tsx|json)\\\\\\"\\"  },  \\"husky\\": {    \\"hooks\\": {      \\"pre-commit\\": \\"lint-staged\\"    }  },  \\"devDependencies\\": {    \\"@namics/prettier-config\\": \\"0.2.0\\",    \\"@namics/tslint-config\\": \\"0.2.0\\",    \\"husky\\": \\"1.2.0\\",    \\"lint-staged\\": \\"8.1.0\\",    \\"npm-run-all\\": \\"4.1.5\\",    \\"prettier\\": \\"1.15.2\\",    \\"tslint\\": \\"5.12.1\\",    \\"tslint-config-prettier\\": \\"1.17.0\\"  }}"`;

exports[`tslint with-prettier-githooks 6`] = `"{	\\"defaultSeverity\\": \\"error\\",	\\"extends\\": [\\"@namics/tslint-config\\", \\"tslint-config-prettier\\"],	\\"jsRules\\": {},	\\"rules\\": {},	\\"rulesDirectory\\": []}"`;
